torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -57,43 +57,59 @@ def are_tensors(var):
 def validate_args_and_maybe_create_graph_inputs(
     sub_args, tracer, tx, manually_set_subgraph_inputs
 ):
-    from . import AutogradFunctionContextVariable, ConstantVariable, TensorVariable
+    from . import (
+        AutogradFunctionContextVariable,
+        ConstantVariable,
+        NNModuleVariable,
+        TensorVariable,
+    )
     from .builder import wrap_fx_proxy
 
     args = []
-    for a in sub_args:
+
+    # One argument to graph per sub_args
+    def add_subarg(subarg):
         assert isinstance(a, VariableTracker)
 
         if isinstance(a, ConstantVariable):
             # Ensures that we recompile when the constant value changes
-            a.add_guard(GuardBuilder.CONSTANT_MATCH)
+            subarg.add_guard(GuardBuilder.CONSTANT_MATCH)
 
             if manually_set_subgraph_inputs:
                 # This arg is not used in the body of the higher order op.
                 # Currently, this new input is added to make the calls
                 # happy, which expect a fixed number of arguments. In
                 # future, we can clean this up.
                 tracer.create_graph_input("const")
-            new_arg = a
-        elif isinstance(a, TensorVariable):
+            new_arg = subarg
+        elif isinstance(subarg, TensorVariable):
             if manually_set_subgraph_inputs:
-                new_proxy = tracer.create_graph_input(a.as_proxy().node.name)
-                example_value = a.as_proxy().node.meta["example_value"]
+                new_proxy = tracer.create_graph_input(subarg.as_proxy().node.name)
+                example_value = subarg.as_proxy().node.meta["example_value"]
                 new_arg = wrap_fx_proxy(
                     tx=tx, proxy=new_proxy, example_value=example_value
                 )
             else:
-                new_arg = a
-        elif isinstance(a, AutogradFunctionContextVariable):
+                new_arg = subarg
+        elif isinstance(subarg, AutogradFunctionContextVariable):
             if manually_set_subgraph_inputs:
-                tracer.create_graph_input(a.as_proxy().node.name)
-            new_arg = a
+                tracer.create_graph_input(subarg.as_proxy().node.name)
+            new_arg = subarg
+        elif not isinstance(
+            subarg, NNModuleVariable
+        ) and subarg.has_unpack_var_sequence(tx):
+            new_arg = []
+            for sub_a in subarg.unpack_var_sequence(tx):
+                new_arg.append(add_subarg(sub_a))
+            new_arg = subarg
```

Comment:
Why do we need this?

If the problem is that HigherOrderOperator body functions don't accept pytree inputs (e.g. a TensorList), the current thinking was that:
- we would not actually record any inputs to the subgraph in this function
- Tensors will automatically get lifted to being an input to the subgraph using the free variable lifting mechanism

This is what the `manually_set_subgraph_inputs=False` flag does (but we'd also need to relax the constraints on the input types, which we haven't done yet)

Reply:
Actually I need to think about this some more...

Reply:
Sorry, say more?